heareval/predictions/task_predictions.py

```diff
@@ -215,9 +307,33 @@ def create_events_from_prediction(
 
     # Convert probabilities to binary vectors based on threshold
     predictions = (predictions > threshold).astype(np.int8)
+    # print("predictions", predictions)
 
+    event_tree = IntervalTree()
+    for label in range(predictions.shape[1]):
+        # print(label, predictions[:, label])
+        # print(np.where(predictions[:, label]))
+        for group in more_itertools.consecutive_groups(
+            np.where(predictions[:, label])[0]
+        ):
```

Comment:
Nice -- this looks clean, wasn't sure how consecutive groups worked -- so this looks for consecutive groups of ones?

Reply:
yes

Reply:
The previous implementation had a bug on the boundaries